File: mobile/lib/features/announcement/domain/entities/announcement.dart
class Announcement {
  final int id;
  final String title;
  final String content;
  final String? image;
  final String? publishedAt;
  final String status;

  const Announcement({
    required this.id,
    required this.title,
    required this.content,
    this.image,
    this.publishedAt,
    required this.status,
  });
}

File: backend/app/Models/Announcement.php
<?php

namespace App\Models;

use App\Traits\BelongsToChurch;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\Factory;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

#[Fillable([
    'title',
    'content',
    'image',
    'published_at',
    'status',
])]
class Announcement extends Model
{
    /** @use HasFactory<Factory<self>> */
    use BelongsToChurch, HasFactory;

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'published_at' => 'datetime',
        ];
    }
}
